versions/3.1.1.md

```diff
@@ -3198,7 +3198,7 @@ Field Name | Type | Applies To | Description
 <a name="securitySchemeScheme"></a>scheme | `string` | `http` | **REQUIRED**. The name of the HTTP Authorization scheme to be used in the [Authorization header as defined in RFC7235](https://tools.ietf.org/html/rfc7235#section-5.1).  The values used SHOULD be registered in the [IANA Authentication Scheme registry](https://www.iana.org/assignments/http-authschemes/http-authschemes.xhtml).  The value is case-insensitive.
 <a name="securitySchemeBearerFormat"></a>bearerFormat | `string` | `http` (`"bearer"`) | A hint to the client to identify how the bearer token is formatted.  Bearer tokens are usually generated by an authorization server, so this information is primarily for documentation purposes.
 <a name="securitySchemeFlows"></a>flows | [OAuth Flows Object](#oauthFlowsObject) | `oauth2` | **REQUIRED**. An object containing configuration information for the flow types supported.
-<a name="securitySchemeOpenIdConnectUrl"></a>openIdConnectUrl | `string` | `openIdConnect` | **REQUIRED**. OpenId Connect URL to discover OAuth2 configuration values. This MUST be in the form of a URL. The OpenID Connect standard requires the use of TLS.
+<a name="securitySchemeOpenIdConnectUrl"></a>openIdConnectUrl | `string` | `openIdConnect` | **REQUIRED**. OpenID Connect URL to discover the [OpenID Connect Configuration Information](https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderConfig). This MUST be in the form of a URL that points to the well-known document and ends with `/.well-known/openid-configuration`. The OpenID Connect standard requires the use of TLS.
```

Comment:
```suggestion
<a name="securitySchemeOpenIdConnectUrl"></a>openIdConnectUrl | `string` | `openIdConnect` | **REQUIRED**. OpenID Connect URL to discover the [OpenID Connect Configuration Information](https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderConfig). This MUST be in the form of a URL. The OpenID Connect standard requires the use of TLS.
```

Could we remove the specifics of the OpenId implementation from this change, I don't think we should duplicate the contents of another spec.

Reply:
Without it's unclear whether you need to provide the issuer or the well-known document. Changing/Removing this would pretty much render this PR useless.

Reply:
@m-mohr the sentiment in the TDC call today was that the `.well-known` requirement is already present in the OpenIDConnect spec, and we normatively cite the spec.  So we already incorporate these, but the advantage of not repeating it is that if OpenIDConnect changes, we do not have to change.  We all agreed that the change to link to the OpenID Connect Configuration Information (instead of "OAuth2 configuration values) was a valuable improvement.  The link also takes you directly to the `.well-known` requirement so that should be clear.

Reply:
I think even without the .well-known part this PR is needed because of the reference  to https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderConfig.
This part `OpenId Connect URL to discover OAuth2 configuration values` was always wrong. OIDC metadata and OAuth2 metadata are not the same.
We, Deutsche Telekom, have a keycloak service where each realm has its own metadata, so openIdConnectUrl cannot be at .well-known URL.
So, [this suggestion](https://github.com/OAI/OpenAPI-Specification/pull/3718#discussion_r1579821713) is not only removing something from another spec, but needed.
Sorry, I did not see this earlier.

Reply:
But then you don't follow the OpenID Connect spec. It requires to use well known part?!

Reply:
The spec says `OpenID Providers supporting Discovery MUST make a JSON document available at the path formed by concatenating the string /.well-known/openid-configuration to the Issuer.`
The Issuer can be e.g. `https://keycloak.example.com/<realm>/` then the openIdConnectUrl would be 
`https://keycloak.example.com/realms/{realm}/.well-known/openid-configuration`.

I think OAI should not care too much, other that the value of openIdConnectUrl should be TLS and point to a JSON document that contains the OP's metadata. e.g.
https://mobileconnect.telekom.de/.well-known/openid-configuration 
https://accounts.login.idm.telekom.com/.well-known/openid-configuration

Reply:
> We, Deutsche Telekom, have a keycloak service where each realm has its own metadata, so openIdConnectUrl cannot be at .well-known URL.

Please read more carefully, your use case is covered. It says "**ends** with `/.well-known/openid-configuration`", not the path must be exactly /.well-known/openid-configuration.

> So we already incorporate the requirement, but the advantage of not repeating it is that if OpenIDConnect changes.

So this means we can even further slim it down, because TLS is also required by the spec and it's a repetition.

```suggestion
<a name="securitySchemeOpenIdConnectUrl"></a>openIdConnectUrl | `string` | `openIdConnect` | **REQUIRED**. Well-known URL to discover the [OpenID Provider Metadata](https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderMetadata).
```

Reply:
I would keep the TLS requirement because I like openapi tools to insist on secure connections. Developers often ignore TLS requirements - I can only speculate why. Please keep the TLS requirement.

Reply:
see comment below